packages/hop-node/src/config/validation.ts

```diff
@@ -225,6 +199,60 @@ export async function validateConfig (config?: FileConfig) {
           if (!minThresholdAmount[token][sourceChain][destinationChain]) {
             throw new Error(`missing minThresholdAmount config for token "${token}" source chain "${sourceChain}" destination chain "${destinationChain}"`)
           }
+        }
+      }
+    }
+  }
+}
+
+export async function validateConfigValues (config?: Config) {
+  if (!config) {
+    throw new Error('config is required')
+  }
+
+  if (!(config instanceof Object)) {
+    throw new Error('config must be a JSON object')
+  }
+
+  for (const chainSlug in config.networks) {
+    const chain = config.networks[chainSlug]
+    if (!chain) {
+      throw new Error(`RPC config for chain "${chain}" is required`)
+    }
+    const { rpcUrl, waitConfirmations } = chain
+    if (!rpcUrl) {
+      throw new Error(`RPC url for chain "${chainSlug}" is required`)
+    }
+    if (typeof rpcUrl !== 'string') {
+      throw new Error(`RPC url for chain "${chainSlug}" must be a string`)
+    }
+    try {
+      const parsed = new URL(rpcUrl)
+      if (!parsed.protocol || !parsed.host || !['http:', 'https:'].includes(parsed.protocol)) {
+        throw new URIError()
+      }
+    } catch (err) {
+      throw new Error(`rpc url "${rpcUrl}" is invalid`)
+    }
+    if (waitConfirmations != null) {
```

Comment:
I was under the impression this was going to be removed from the config validation since it is not a part of the config file. Is this `validateConfigValues` function used for more than validating just the config file?

Reply:
> Is this validateConfigValues function used for more than validating just the config file?

That is correct, it validates the in memory [`config`](https://github.com/hop-protocol/hop/blob/c58a99169aa6c08f3fbcba11b4d8586b909d3e62/packages/hop-node/src/config/config.ts#L125) object which has some values loaded by default (like `waitConfirmations`) so it would catch any misconfiguration values from the core package or values from config file